eng/pipelines/templates/steps/sync-repo.yml

```diff
@@ -0,0 +1,43 @@
+parameters:
+  # Provde a list of repos
+  # Example:
+  #   Azure/azure-rest-api-specs:
+  #     Branch: main
+  #     TargetRepos:
+  #       Azure/azure-rest-api-specs-pr:
+  #       azure-sdk/azure-rest-api-specs:
+  #         Branch: dev
+  #         Rebase: true
+  #   Azure/azure-sdk-for-go:
+  #     Branch: main
+  #     TargetRepos:
+  #       Azure/azure-sdk-for-go-pr:
+  - name: Repos
+    type: object
+  - name: ScriptDirectory
+    type: string
+    default: eng/scripts
+  - name: WorkingDirectory
+    type: string
+    default: $(System.DefaultWorkingDirectory)
+
+steps:
+
+- ${{ each repo in parameters.Repos }}:
+
+  - ${{ each target in repo.value.TargetRepos }}:
+
+    - task: PowerShell@2
+      displayName: Sync Repo from ${{ repo.key }} to ${{ target.key }}
+      inputs:
+        pwsh: true
+        workingDirectory: ${{ parameters.WorkingDirectory }}
+        filePath: ${{ parameters.ScriptDirectory }}/Git-Repo-Sync.ps1
+        arguments: >
+          -SourceRepo '${{ repo.key }}'
+          -SourceBranch '${{ repo.value.Branch }}'
+          -TargetRepo '${{ target.key }}'
+          -TargetBranch '${{ coalesce(target.value.Branch, repo.value.Branch) }}'
+          -Rebase '${{ target.value.Rebase }}'
+          -PushArgs "${{ target.value.PushArgs }}"
```

Comment:
Do you have cases where we would need this yet? If not I would remove it and only add it when we have a clear use case.

Reply:
Can you point me at a place where we need to pass in these extra PushArgs?

Reply:
@weshaggard No, I deleted this parameter